### ISSUE TITLE
failed reproduce llama3-8b result 

### ISSUE THREAD

**Issue by chunniunai220ml** [OpenLMLab/LEval#14]
i can not reproduce llama3-8b result according ur advice, just got
{'exact_match': 53.9604, 'num_predicted': 202, 'mean_prediction_length_characters': 1.0, 'LEval_score': 53.9604, 'display_keys': ['exact_match'], 'display': [53.9604]}

here is my codes:
python Baselines/llama2-chat-test.py
--metric exam_eval
--task_name quality
--max_length 4k

and change llama2-chat-test.py
elif args.metric == "exam_eval":
context = "Document is as follows. {document} \nQuestion: {inst}. Please directly give the answer without any additional output or explanation "

message="<|begin_of_text|>"+sys_prompt    # B_INST + B_SYS + sys_prompt + E_SYS + context + E_INST
message += "\nAnswer:"

**Comment by ChenxinAn-fdu** [comment]
Hi!  You should use the `Instruct` version of Llama3 8B and set max_length to 8k. 
Please use the chat template of Llama3.

**Comment by chunniunai220ml** [comment]
thank for ur reply, i modified the code according ur advice, as follows:

![image](https://github.com/OpenLMLab/LEval/assets/17668109/78f31233-9aeb-442c-babb-9e40550775ee)

so my message is : 

<|begin_of_text|><|start_header_id|>system<|end_header_id|>Now you are given a very long document. Please follow the instruction based on this document. For multi-choice questions, there is only a sinlge correct option. Please only provide the letter corresponding to the answer (like A or B) when answering. For other questions, please directly give the concise and accurate answer. <|eot_id|><|start_header_id|>user<|end_header_id|>Document is as follows. {document}
Question: {inst}.  Please directly give the answer without any additional output or explanation <|eot_id|><|start_header_id|>assistant<|end_header_id|>

then i got ：
{'exact_match': 63.8614, 'num_predicted': 202, 'mean_prediction_length_characters': 1.0, 'LEval_score': 63.8614, 'display_keys': ['exact_match'], 'display': [63.8614]}
a little different with ur report, is normal?   i did not use  llama_flash_attn_monkey_patch

**Comment by ylsung** [comment]
Hi,

Thank you for providing the codes and tips for reproducing LLaMA 3 results!

I modified the LLaMA 2 codes based on your suggestions:
1. Use the LLaMA3-Instruct model
2. Set the max_length to 8k
3. Use the llama3 template (as shown below)
```python
message = ""
message += "<|begin_of_text|><|start_header_id|>system<|end_header_id|>"
message += "\n" + sys_prompt
message += "<|eot_id|><|start_header_id|>user<|end_header_id|>"
message += "\n" + context
message += "<|eot_id|><|start_header_id|>assistant<|end_header_id|>"
```

The results I got for the six tasks are

|  Llama3-8b | TOEFL | QuALITY | Coursera | SFiction | GSM | CodeU |
|--------|------|------|-------|-------|-------|-------|
| Your Results | 82.89 | 64.85 |  53.77 | 69.53 |  79.00 | 2.22|
| My Reproduction | 81.04 | 61.88 |  52.62 | 71.09 |  _**29.00**_ | 4.44|

Results on most datasets are within an acceptable gap to your results while the GSM100k result I got is somehow very bad.
Could you please help me check if my prompt is the same as yours? Or do you have any other suggestions for reproducing the results (such as tuning the decoding hyperparameters)? Thank you very much.

**Comment by ChenxinAn-fdu** [comment]
Hi! I suggest using:
```
message = ""
message += "<|begin_of_text|><|start_header_id|>system<|end_header_id|>"
message += "\n" + sys_prompt
message += "<|eot_id|><|start_header_id|>user<|end_header_id|>"
message += "\n" + context + "\nAnswer:"
```
The `Question` and `Answer` pair is needed to achieve high performance on math tasks.

**Comment by ylsung** [comment]
Thanks for your reply.

I found the role special tokens have to be added to all the examples in GSM100k, such as 

```python
context = document + "\n\n" + inst
                    
context = context.replace(
   "Question:", 
   "<|eot_id|><|start_header_id|>user<|end_header_id|>\nQuestion:"
)
                    
context = context.replace(
    "Let's think step by step", 
    "Let's think step by step\n<|eot_id|><|start_header_id|>assistant<|end_header_id|>"
)
                    
message = ""
message += "<|begin_of_text|><|start_header_id|>system<|end_header_id|>"
message += "\n" + sys_prompt
message += context
```

Then the accuracy will be 78!

There is also the other option not to use any chat format

```python
message = sys_prompt + "\n" + context
```

the model will act like a pre-trained language model and keep outputting self-curated questions and answers after the CoT and answer for the original question. If we parse the first answer that the model generates (which has been done in the current code), the accuracy is 80.
